ci/purge.ps1

```diff
@@ -0,0 +1,33 @@
+# This script is used to kill all dangling procceses at the end of a Buildkite build.
+# 
+# It does this by using [handle](https://docs.microsoft.com/en-us/sysinternals/downloads/handle)
+# to find all processes of a particular name which have file locks on our build (project) directory.
+# We can then attempt to kill each of these processes if any exist.
+
+$SCRIPT_LOCATION = $MyInvocation.MyCommand.Path
+$SCRIPT_DIR = Split-Path $SCRIPT_LOCATION
+cd "${SCRIPT_DIR}/.."
+
+function Kill-Dangling-Processes {
+	param( [string]$ProcessName )
+
+	echo "Looking for ${ProcessName} processes to kill..."
+	${DIR}=$(pwd).Path
+	${OUT}=$(handle64 -accepteula -p "${ProcessName}.exe" ${DIR})
```

Comment:
I recommend avoiding using process-names as the only filter, since when Windows nodes have more than one buildkite-agent process running builds, this will interfere with other running builds.

(It's only an accident of fate that has limited BK to one process for Windows nodes - we haven't built the multi-agent per node support yet on Windows, but will, though you'll still have the option of how many to run).

Reply:
Discussed offline - this does currently filter by build directory and shouldn't wreak havoc on a multi-agent per node setup.